pkg/plot/plot.go

```diff
@@ -115,27 +115,27 @@ func (series *Series) Scale(factor Value) {
 // into the Summary map.
 func (series *Series) Summarize(percentiles []float64) {
 	var (
-		min, max, total Value
-		nValidPlots     int64
-		nPlots          = len(series.Plots)
+		min, max, total, current Value
+		nValidPlots              int64
+		nPlots                   = len(series.Plots)
 	)
 
 	if nPlots > 0 {
 		min = series.Plots[0].Value
-		series.Summary["last"] = series.Plots[nPlots-1].Value
 	}
 
 	for i := range series.Plots {
-		if !series.Plots[i].Value.IsNaN() && series.Plots[i].Value < min || min.IsNaN() {
-			min = series.Plots[i].Value
-		}
+		if !series.Plots[i].Value.IsNaN() {
+			current = series.Plots[i].Value
+			if current < min || min.IsNaN() {
+				min = series.Plots[i].Value
+			}
 
-		if series.Plots[i].Value > max {
-			max = series.Plots[i].Value
-		}
+			if current > max {
```

Comment:
Maybe we need to add a `} else if …` to prevent even looking at the second if condition.